src/coreclr/jit/optimizer.cpp

```diff
@@ -2687,19 +2688,23 @@ void Compiler::optFindNaturalLoops()
         fgUpdateChangedFlowGraph(FlowGraphUpdates::COMPUTE_DOMS);
     }
 
-    if (false /* pre-header stress */)
+    // Create loop pre-header for every loop.
+    bool modForPreHeader = false;
+    for (unsigned loopInd = 0; loopInd < optLoopCount; loopInd++)
     {
-        // Stress mode: aggressively create loop pre-header for every loop.
-        for (unsigned loopInd = 0; loopInd < optLoopCount; loopInd++)
+        if (fgCreateLoopPreHeader(loopInd))
         {
-            fgCreateLoopPreHeader(loopInd);
-        }
-
-        if (fgModified)
-        {
-            fgUpdateChangedFlowGraph(FlowGraphUpdates::COMPUTE_DOMS);
+            modForPreHeader = true;
         }
     }
+    if (modForPreHeader)
+    {
+        // The predecessors were maintained in fgCreateLoopPreHeader; don't rebuild them.
```

Comment:
I am confused by this comment. We never rebuild predecessors anymore (at least not globally).

Reply:
I wrote this comment before we stopped rebuilding preds